### PR TITLE
Details Comp

### PR DESCRIPTION
**Pull request by flore77**
## Need

```
As a developer
I need a details component
So that I can display the screenshots
```

## Deliverables

A component that displays the baseline/diff/screenshot.

## Solution

The Test comp will render this one, which will contain 3 images, with the links provided from `Mugshot's result`. 

The `Details` comp will received via props the `paths` of the images.

Tests:

Details comp:
* should render the baseline
* should render the diff if there is one
* should not render the diff if there is none
* should render the screenshot if there is one
* should not render the screenshot if there is none

Test comp:
* should pass the paths to the Details comp

### CODE REVIEW

**Comment by flore77** [comment]
@andrei-picus-hs please review. 

I also added 3 new tests:

* should have the correct baseline/diff/screenshot path

File: tests/unit/components/details.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  it('should render the baseline', function() {
+    tree = sd.shallowRender(<Details paths={passDetails}/>);
+
+    expect(tree.findNode('.baseline')).to.not.be.false;
+  });
+
+  describe('Pass details', function() {
+    it('should not render the diff', function() {
+      tree = sd.shallowRender(<Details paths={passDetails}/>);
```
Comment (andrei-picus-hs) [inline tests/unit/components/details.js]: This is duplicated code and should sit in a `beforeEach` block. You should have a `describe` block that contains all the tests that are suitable for passed and failed tests + the ones that apply only to pass tests. Then, in another `describe` block, place the tests that apply only to failed tests.

File: ui/components/details.jsx [hunk 2]
```diff
@@ -2,11 +2,21 @@ var React = require('react');
...
   render: function() {
-    return <div className="details">
-      <img src="http://placehold.it/100"/>
-      <img src="http://placehold.it/100"/>
-      <img src="http://placehold.it/100"/>
-    </div>
+    var paths = this.props.paths,
+        imgs = [];
+
+    imgs.push(<img className='baseline' src={paths.baseline}
```
Comment (andrei-picus-hs) [inline ui/components/details.jsx]: Put the baseline right in the declaration above.

**Comment by andrei-picus-hs** [comment]
Finished review.

**Comment by flore77** [comment]
@andrei-picus-hs please review

File: tests/unit/components/details.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  describe('Passing', function() {
+    it('should render the baseline', function() {
+      tree = sd.shallowRender(<Details paths={passDetails}/>);
+
+      expect(tree.findNode('.baseline')).to.not.be.false;
+    });
+
+    it('should not render the diff', function() {
+      tree = sd.shallowRender(<Details paths={passDetails}/>);
```
Comment (andrei-picus-hs) [inline tests/unit/components/details.js]: Code duplication. Move this into a `beforeEach`.

File: tests/unit/components/details.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  describe('Fail details', function() {
+    it('should render the diff', function() {
+      tree = sd.shallowRender(<Details paths={failDetails}/>);
+
+      expect(tree.findNode('.diff')).to.not.be.false;
+    });
+
+    it('should render the screenshot', function() {
+      tree = sd.shallowRender(<Details paths={failDetails}/>);
```
Comment (andrei-picus-hs) [inline tests/unit/components/details.js]: Same here.

File: tests/unit/components/details.js (hunk 3 above)
Comment (flore77) [inline tests/unit/components/details.js]: sorry forgot about it :disappointed:

File: tests/unit/components/test.js [hunk 5]
```diff
@@ -47,6 +55,16 @@ describe('Test', function() {
...
     });
+
+    it('should pass the failDetails to the Details component', function() {
+      var expected = {
+        baseline: failTest.result.baseline,
+        diff: failTest.result.diff,
+        screenshot: failTest.result.screenshot
+      };
+
+      expect(tree.findNode('Details').props.paths).to.be.deep.equal(expected);
```
Comment (andrei-picus-hs) [inline tests/unit/components/test.js]: You're checking the baseline here and you're checking it again in one of the above tests. This tells me something: that you're tests are not focused, they test more than they can chew. So, write a test for a single property. You only need to check the baseline once (in the pass test describe).

**Comment by andrei-picus-hs** [comment]
Finished.

**Pull request merged**